src/main/java/org/springframework/social/openidconnect/HttpClientFactory.java

```diff
@@ -114,4 +116,26 @@ private static X509HostnameVerifier getVerifier(boolean isStrict){
         return SSLSocketFactory.STRICT_HOSTNAME_VERIFIER;
     }
 
+    /**
+     * Gets the TLS version from jdk parameter.  Defaults to version 1.2 for JDK 1.7 if not provided
+     * @return
+     */
+    public static String getTlsVersion() {
+        // don't use "jdk.tls.client.protocols" as it was causing unexpected exception during unit testing
+        String tlsVer = System.getProperty("tlsProtocol");
+
+        if(StringUtils.isEmpty(tlsVer)) {
+            double javaVersion = Double.parseDouble(System.getProperty("java.specification.version"));
+            if(javaVersion == 1.7) {
```

Comment:
I think JDK 1.7 is the only case where you have both TLS 1.2 and order version. In that i can i'm fine with checking the logic for specific JDK version.